libindy/src/commands/metrics.rs

```diff
@@ -12,18 +29,585 @@ const OPENED_WALLET_IDS_COUNT: &str = "opened_wallet_ids_count";
+    WalletCommandGenerateKey,
+    WalletCommandDeriveKey,
+    // PairwiseCommand
+    PairwiseCommandPairwiseExists,
+    PairwiseCommandCreatePairwise,
+    PairwiseCommandListPairwise,
+    PairwiseCommandGetPairwise,
+    PairwiseCommandSetPairwiseMetadata,
+    // NonSecretsCommand
+    NonSecretsCommandAddRecord,
+    NonSecretsCommandUpdateRecordValue,
+    NonSecretsCommandUpdateRecordTags,
+    NonSecretsCommandAddRecordTags,
+    NonSecretsCommandDeleteRecordTags,
+    NonSecretsCommandDeleteRecord,
+    NonSecretsCommandGetRecord,
+    NonSecretsCommandOpenSearch,
+    NonSecretsCommandFetchSearchNextRecords,
+    NonSecretsCommandCloseSearch,
+    // PaymentsCommand
+    PaymentsCommandRegisterMethod,
+    PaymentsCommandCreateAddress,
+    PaymentsCommandCreateAddressAck,
+    PaymentsCommandListAddresses,
+    PaymentsCommandAddRequestFees,
+    PaymentsCommandAddRequestFeesAck,
+    PaymentsCommandParseResponseWithFees,
+    PaymentsCommandParseResponseWithFeesAck,
+    PaymentsCommandBuildGetPaymentSourcesRequest,
+    PaymentsCommandBuildGetPaymentSourcesRequestAck,
+    PaymentsCommandParseGetPaymentSourcesResponse,
+    PaymentsCommandParseGetPaymentSourcesResponseAck,
+    PaymentsCommandBuildPaymentReq,
+    PaymentsCommandBuildPaymentReqAck,
+    PaymentsCommandParsePaymentResponse,
+    PaymentsCommandParsePaymentResponseAck,
+    PaymentsCommandAppendTxnAuthorAgreementAcceptanceToExtra,
+    PaymentsCommandBuildMintReq,
+    PaymentsCommandBuildMintReqAck,
+    PaymentsCommandBuildSetTxnFeesReq,
+    PaymentsCommandBuildSetTxnFeesReqAck,
+    PaymentsCommandBuildGetTxnFeesReq,
+    PaymentsCommandBuildGetTxnFeesReqAck,
+    PaymentsCommandParseGetTxnFeesResponse,
+    PaymentsCommandParseGetTxnFeesResponseAck,
+    PaymentsCommandBuildVerifyPaymentReq,
+    PaymentsCommandBuildVerifyPaymentReqAck,
+    PaymentsCommandParseVerifyPaymentResponse,
+    PaymentsCommandParseVerifyPaymentResponseAck,
+    PaymentsCommandGetRequestInfo,
+    PaymentsCommandSignWithAddressReq,
+    PaymentsCommandSignWithAddressAck,
+    PaymentsCommandVerifyWithAddressReq,
+    PaymentsCommandVerifyWithAddressAck,
+    // CacheCommand
+    CacheCommandGetSchema,
+    CacheCommandGetSchemaContinue,
+    CacheCommandGetCredDef,
+    CacheCommandGetCredDefContinue,
+    CacheCommandPurgeSchemaCache,
+    CacheCommandPurgeCredDefCache,
+    // MetricsCommand
+    MetricsCommandCollectMetrics,
+    // Exit
+    Exit,
+}
+
+
 pub enum MetricsCommand {
     CollectMetrics(Box<dyn Fn(IndyResult<String>) + Send>)
 }
 
+const COMMANDS_COUNT: usize = MetricsStorage::commands_count();
+
+pub struct MetricsStorage {
+    queue_count: [u64; COMMANDS_COUNT],
+    queue_duration: [u64; COMMANDS_COUNT],
+
+    execute_count: [u64; COMMANDS_COUNT],
+    execute_duration: [u64; COMMANDS_COUNT],
+}
+
+impl MetricsStorage {
+    pub fn new() -> MetricsStorage {
+        MetricsStorage {
+            queue_count: [u64::MIN; COMMANDS_COUNT],
```

Comment:
My Rust knowledge is a little bit rusty, so I might be wrong here, but I thought it should be something like `[u64; COMMANDS_COUNT]`, why `MIN` here? 

FYI @jovfer

Reply:
I also doubt my Rust knowledge. But this is creating an array of zeros.